roles/kubernetes/preinstall/tasks/0010-swapoff.yml

```diff
@@ -13,11 +13,17 @@
   command: /sbin/swapon -s
   register: swapon
   changed_when: no
+
 - name: Disable swap
   command: /sbin/swapoff -a
-  when: swapon.stdout
+  when:
+    - swapon.stdout
+    - kubelet_fail_swap_on | default(True)
   ignore_errors: "{{ ansible_check_mode }}"  # noqa ignore-errors
 
 - name: Disable swapOnZram for Fedora
   command: touch /etc/systemd/zram-generator.conf
-  when: swapon.stdout and ansible_distribution in ['Fedora']
+  when:
+    - swapon.stdout
+    - ansible_distribution in ['Fedora']
+    - kubelet_fail_swap_on | default(True)
```

Comment:
Shouldn't this be added to default instead of being a secret/hidden feature 😄 ?

Reply:
Considering this is still alpha in upstream I don't think it's worth exposing it until it graduates to beta at least.